src/osd/osd_types.cc

```diff
@@ -5878,3 +5878,33 @@ ostream& operator<<(ostream& out, const store_statfs_t &s)
       << ")";
   return out;
 }
+
+void OSDOp::clear_data(vector<OSDOp>& ops)
+{
+  for (unsigned i = 0; i < ops.size(); i++) {
+    OSDOp& op = ops[i];
+    op.outdata.clear();
+    if (ceph_osd_op_type_attr(op.op.op) &&
+        op.op.xattr.name_len &&
+	op.indata.length() >= op.op.xattr.name_len) {
+      bufferptr bp(op.op.xattr.name_len);
+      bufferlist bl;
+      bl.append(bp);
+      bl.copy_in(0, op.op.xattr.name_len, op.indata);
+      op.indata.claim(bl);
```

Comment:
I think the idea is that you only keep enough buffer around to render operator<<, but can free the rest of the memory.

This whole problem is annoying.  A simple solution would just be to render the operator<< output as a string and store that in OpRequest, but it turns out it is the slowest part (or close to it) of the whole optracker.  And most of the time nobody queries this info anyway.

TBH this method looks pretty slow too.. I'm not sure the slowdown here isn't too much that we're better off with the (yes, annoying) status quo?